testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/suspend/SuspendOnSoftKillTestCase.java

```diff
@@ -0,0 +1,414 @@
+/*
+Copyright 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package org.jboss.as.test.manualmode.suspend;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.NAME;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.READ_ATTRIBUTE_OPERATION;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUSPEND_STATE;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.SocketPermission;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+
+import org.jboss.as.test.integration.common.HttpRequest;
+import org.jboss.as.test.shared.PermissionUtils;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jboss.as.test.shared.TimeoutUtil;
+import org.jboss.dmr.ModelNode;
+import org.jboss.logging.Logger;
+import org.jboss.msc.service.ServiceActivator;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.AssumptionViolatedException;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.core.testrunner.ServerControl;
+import org.wildfly.core.testrunner.ServerController;
+import org.wildfly.core.testrunner.WildflyTestRunner;
+import org.wildfly.test.suspendresumeendpoint.SuspendResumeHandler;
+import org.wildfly.test.suspendresumeendpoint.TestSuspendServiceActivator;
+import org.wildfly.test.suspendresumeendpoint.TestUndertowService;
+import org.xnio.IoUtils;
+
+/**
+ * WFCORE-3073. Tests that a softly killed process shuts down gracefully.
+ *
+ * @author Brian Stansberry
+ */
+@RunWith(WildflyTestRunner.class)
+@ServerControl(manual = true)
+public class SuspendOnSoftKillTestCase {
+
+    private static final int TIMEOUT = 20000;
```

Comment:
Is this value correct?
I suppose you mean 20000ms (20 seconds) but the code in `BootstrapImpl` will compute it as 20000 seconds

Reply:
@jmesnil This value isn't passed to the server for use by BootstrapImpl; it's just the # of ms used in waitForServerShutdown() when the test is waiting for the server to completely stop after the test of suspend behavior is done.

I'll clean this up a bit; get rid of the constant so it's less confusing, use TimeoutUtil more consistently etc.